contracts/bridge/Bridge.sol

```diff
@@ -207,7 +235,56 @@ contract Bridge is ITokenReceiver, INFTReceiver, Ownable {
             msg.value,
             address(0),
             _to,
-            requestNonce
+            requestNonce,
+            ""
+        );
+        requestNonce++;
+    }
+
+    // requestERC20Transfer requests transfer ERC20 to _to on relative chain.
+    function requestERC20Transfer(address _contractAddress, address _to, uint256 _amount) external {
+        require(isRunning, "stopped bridge");
+        require(_amount > 0, "zero msg.value");
+        require(allowedTokens[_contractAddress] != address(0), "Not a valid token");
+
+        if (modeMintBurn) {
+            ERC20Burnable(_contractAddress).burnFrom(msg.sender, _amount);
+        } else {
+            IERC20(_contractAddress).transferFrom(msg.sender, address(this), _amount);
+        }
+
+        emit RequestValueTransfer(
+            TokenKind.ERC20,
+            msg.sender,
+            _amount,
+            _contractAddress,
+            _to,
+            requestNonce,
+            ""
+        );
+        requestNonce++;
+    }
+
+    // requestERC721Transfer requests transfer ERC721 to _to on relative chain.
+    function requestERC721Transfer(address _contractAddress, address _to, uint256 _uid) external {
+        require(isRunning, "stopped bridge");
+        require(allowedTokens[_contractAddress] != address(0), "Not a valid token");
+
+        string memory uri = ERC721Metadata(_contractAddress).tokenURI(_uid);
+
+        IERC721(_contractAddress).transferFrom(msg.sender, address(this), _uid);
```

Comment:
The conditions are different from the ERC20, is there any particular reason?

Reply:
Right. In ERC721, there is not `burnFrom`. So this code was written.
In wemade code, they also used like this way.